dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java

```diff
@@ -388,50 +387,77 @@ private void deleteSecrets(final String key, final String siteIdentifier, final
+                    "Invalid attempt to create an app descriptor performed by user with id `%s`.",
                     user.getUserId()));
         }
-
         final String ymlFilesPath = getServiceDescriptorDirectory();
         final File basePath = new File(ymlFilesPath);
         if (!basePath.exists()) {
-            basePath.mkdir();
-        }
-        Logger.debug(AppsAPIImpl.class,
-                () -> " ymlFiles are set under:  " + ymlFilesPath);
-
-        // Now validate the incoming file.. see if we're rewriting an existing file or attempting to re-use an already in use service-key.
-        final AppDescriptor serviceDescriptor = ymlMapper
-                .readValue(inputStream, AppDescriptor.class);
+            basePath.mkdirs();
+        }
+        Logger.debug(AppsAPIImpl.class, () -> " ymlFiles are set under:  " + ymlFilesPath);
+
+            final AppSchema appSchema = readAppFile(file);
+            // Now validate the incoming file.. see if we're rewriting an existing file or attempting to re-use an already in use service-key.
+            if (validateServiceDescriptor(appSchema)) {
+                final File incomingFile = new File(basePath, file.getName());
+                if (incomingFile.exists()) {
+                    throw new DotDataException(
+                            String.format(
+                                    "Invalid attempt to override an existing file named '%s'.",
+                                    incomingFile.getName()));
+                }
 
-        if (validateServiceDescriptor(serviceDescriptor)
-                && validateAppDescriptorUniqueName(serviceDescriptor)) {
+                writeAppFile(incomingFile, appSchema);
 
-            final String serviceKey = serviceDescriptor.getKey();
-            final File incomingFile = new File(basePath, String.format("%s.yml", serviceKey));
-            if (incomingFile.exists()) {
-                throw new DotDataException(
-                        String.format("Invalid attempt to override an existing file named '%s'.",
-                                incomingFile.getName()));
+                invalidateCache();
             }
+            return new AppDescriptorImpl(file.getName(), appSchema);
 
-            ymlMapper.writeValue(incomingFile, serviceDescriptor);
+    }
 
-            invalidateCache();
+    /**
+     * There's a version of the method readValue on the ymlMapper which takes a file and internally creates directly a FileInputStream
+     * According to https://dzone.com/articles/fileinputstream-fileoutputstream-considered-harmful
+     * that's very harmful
+     * @param file
+     * @return
+     * @throws DotDataException
+     */
+    private AppSchema readAppFile(final File file) throws DotDataException {
+        try (final InputStream inputStream = Files.newInputStream(Paths.get(file.getPath()))) {
+            return ymlMapper.readValue(inputStream, AppSchema.class);
+        }catch (Exception e){
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5745882)